File: src/main/java/io/github/wntopia/gikipedia/server/domain/auth/dto/response/DataGsmUserInfoResDto.java
package io.github.wntopia.gikipedia.server.domain.auth.dto.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;

@JsonIgnoreProperties(ignoreUnknown = true)
public record DataGsmUserInfoResDto(
    Long id,
    String email,
    String role,
    @JsonProperty("isStudent") Boolean isStudent,
    DataGsmStudentInfoResDto student)
    implements Serializable {}

File: src/main/kotlin/io/github/wntopia/gikipedia/server/domain/auth/dto/response/DataGsmUserInfoResDto.kt
package io.github.wntopia.gikipedia.server.domain.auth.dto.response

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
data class DataGsmUserInfoResDto(
    val id: Long?,
    val email: String?,
    val role: String?,
    @JsonProperty("isStudent") val isStudent: Boolean?,
    val student: DataGsmStudentInfoResDto?,
) : Serializable
